javaslang/src/test/java/javaslang/collection/AbstractSeqTest.java

```diff
@@ -1217,6 +1217,27 @@ public void shouldCreateReverseIteratorOfNonEmpty() {
         assertThat(ofAll(of("a", "b", "c")).reverseIterator().toList()).isEqualTo(Iterator.of("c", "b", "a").toList());
     }
 
+    // -- shuffle
+
+    @Test
+    public void shouldShuffleEmpty() {
+        assertThat(empty().shuffle().isEmpty());
+    }
+
+    @Test
+    public void shouldShuffleHaveSameLength() {
+        assertThat(of(1, 2, 3).shuffle().size()).isEqualTo(of(1, 2, 3).size());
+    }
+
+    @Test
+    public void shouldShuffleHaveSameElements() {
+        Seq<Integer> shuffled = of(1, 2, 3).shuffle();
+        assertThat(shuffled.indexOf(1)).isNotEqualTo(-1);
```

Comment:
@io7m @paplorinc for me the code is perfectly ok. Limiting the scope is a well thought-through decision.

@io7m Please add shuffle() also to `IndexedSeq` and I can merge this PR!